File: ruby/pangram/pangram.rb
class Pangram
  VERSION = 1

  def self.is_pangram?(string)
    new(string).is_pangram?
  end

  def initialize(string)
    @string = string.downcase
    @alphabet = ("a".."z").to_a
  end

  def is_pangram?
    @string.each_char do |char|
      @alphabet.delete(char)
    end
    @alphabet.length == 0
  end
end

File: elixir/pangram/pangram.exs
defmodule Pangram do
  @letters ~w(a b c d e f g h i j k l m n o p q r s t u v w x y z)
  @alphabet Enum.join(@letters)

  @spec pangram?(String.t) :: boolean
  def pangram?(sentence) do
    sentence
    |> String.downcase
    |> String.codepoints
    |> Enum.filter(&(&1 in @letters))
    |> Enum.uniq
    |> Enum.sort
    |> Enum.join == @alphabet
  end
end
